pkg/p2p/libp2p/libp2p.go

```diff
@@ -708,35 +708,27 @@ func buildUnderlayAddress(addr ma.Multiaddr, peerID libp2ppeer.ID) (ma.Multiaddr
 	return addr.Encapsulate(hostAddr), nil
 }
 
-func (s *Service) Connect(ctx context.Context, addr ma.Multiaddr) (address *bzz.Address, err error) {
+func (s *Service) Connect(ctx context.Context, addrs []ma.Multiaddr) (address *bzz.Address, err error) {
 	loggerV1 := s.logger.V(1).Register()
 
 	defer func() {
 		err = s.determineCurrentNetworkStatus(err)
 	}()
 
-	// Extract the peer ID from the multiaddr.
-	info, err := libp2ppeer.AddrInfoFromP2pAddr(addr)
+	id, remotes, err := normalizeAddrsSamePeer(addrs)
 	if err != nil {
-		return nil, fmt.Errorf("addr from p2p: %w", err)
-	}
-
-	hostAddr, err := buildHostAddress(info.ID)
-	if err != nil {
-		return nil, fmt.Errorf("build host address: %w", err)
+		return nil, err
 	}
 
-	remoteAddr := addr.Decapsulate(hostAddr)
-
-	if overlay, found := s.peers.isConnected(info.ID, remoteAddr); found {
-		address = &bzz.Address{
-			Overlay:  overlay,
-			Underlay: addr,
+	for _, r := range remotes {
+		if overlay, found := s.peers.isConnected(id, r); found {
+			return &bzz.Address{Overlay: overlay, Underlay: addrs}, p2p.ErrAlreadyConnected
 		}
-		return address, p2p.ErrAlreadyConnected
 	}
 
-	if err := s.connectionBreaker.Execute(func() error { return s.host.Connect(ctx, *info) }); err != nil {
+	// libp2p will chose first ready address (possible multi dial).
+	info := libp2ppeer.AddrInfo{ID: id, Addrs: remotes}
```

Comment:
`remotes` that are returned from the normalizeAddrsSamePeer have "decapsulated" hostAddr. Are such multiadds ok (complete) to be set as AddrInfo.Addrs?